zetaclient/chains/solana/observer/outbound_test.go

```diff
@@ -0,0 +1,281 @@
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/base"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/solana/observer"
+	"github.com/zeta-chain/zetacore/zetaclient/db"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils/mocks"
+)
+
+const (
+	// gatewayAddressDevnet is the gateway address on devnet for testing
+	GatewayAddressTest = "2kJndCL9NBR36ySiQ4bmArs4YgWQu67LmCDfLzk5Gb7s"
+
+	// withdrawTxTest is an archived withdraw tx result on devnet for testing
+	// https://explorer.solana.com/tx/5iBYjBYCphzjHKfmPwddMWpV2RNssmzk9Z8NNmV9Rei71pZKBTEVdkmUeyXfn7eWbV8932uSsPfBxgA7UgERNTvq?cluster=devnet
+	withdrawTxTest = "5iBYjBYCphzjHKfmPwddMWpV2RNssmzk9Z8NNmV9Rei71pZKBTEVdkmUeyXfn7eWbV8932uSsPfBxgA7UgERNTvq"
+
+	// withdrawFailedTxTest is an archived failed withdraw tx result on devnet for testing
+	// https://explorer.solana.com/tx/5nFUQgNSdqTd4aPS4a1xNcbehj19hDzuQLfBqFRj8g7BJdESVY6hFuTFPWFuV6aWAfzEMfVfCdNu9DfzVp5FsHg5?cluster=devnet
+	withdrawFailedTxTest = "5nFUQgNSdqTd4aPS4a1xNcbehj19hDzuQLfBqFRj8g7BJdESVY6hFuTFPWFuV6aWAfzEMfVfCdNu9DfzVp5FsHg5"
+
+	// tssAddressTest is the TSS address for testing
+	tssAddressTest = "0x05C7dBdd1954D59c9afaB848dA7d8DD3F35e69Cd"
+)
+
+// createTestObserver creates a test observer for testing
+func createTestObserver(
+	t *testing.T,
+	chain chains.Chain,
+	solClient interfaces.SolanaRPCClient,
+	tss interfaces.TSSSigner,
+) *observer.Observer {
+	database, err := db.NewFromSqliteInMemory(true)
+	require.NoError(t, err)
+
+	// create observer
+	chainParams := sample.ChainParams(chain.ChainId)
+	chainParams.GatewayAddress = GatewayAddressTest
+	ob, err := observer.NewObserver(chain, solClient, *chainParams, nil, tss, database, base.DefaultLogger(), nil)
+	require.NoError(t, err)
+
+	return ob
+}
+
+func Test_CheckFinalizedTx(t *testing.T) {
+	// the test chain and transaction hash
+	chain := chains.SolanaDevnet
+	txHash := withdrawTxTest
+	txHashFailed := withdrawFailedTxTest
+	txSig := solana.MustSignatureFromBase58(txHash)
+	coinType := coin.CoinType_Gas
+	nonce := uint64(0)
+
+	// load archived outbound tx result
+	txResult := testutils.LoadSolanaOutboundTxResult(t, TestDataDir, chain.ChainId, txHash)
+
+	// mock GetTransaction result
+	solClient := mocks.NewSolanaRPCClient(t)
+	solClient.On("GetTransaction", mock.Anything, txSig, mock.Anything).Return(txResult, nil)
+
+	// mock TSS
+	tss := mocks.NewMockTSS(chain, tssAddressTest, "")
+
+	// create observer with and TSS
+	ob := createTestObserver(t, chain, solClient, tss)
+	ctx := context.TODO()
+
+	t.Run("should successfully check finalized tx", func(t *testing.T) {
+		tx, finalized := ob.CheckFinalizedTx(ctx, txHash, nonce, coinType)
+		require.True(t, finalized)
+		require.NotNil(t, tx)
+	})
+	t.Run("should return error on invalid tx hash", func(t *testing.T) {
```

Comment:
```suggestion
	})
	
	t.Run("should return error on invalid tx hash", func(t *testing.T) {
```
Can we leave a space between Runs (same comment for all tests below)

Reply:
https://github.com/zeta-chain/node/pull/2560/commits/ada365437e0782d6d2f6725ca87d6c53a687118d